### PR TITLE
Update API versions to 56.0 and test classes to use new Assert class

### PR DESCRIPTION
**Pull request by chazwatkins**
This change is [Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/force-di/88)

### CODE REVIEW

File: force-di/test/classes/di_BindingParamTest.cls [hunk 1]
```diff
@@ -44,19 +44,19 @@ private class di_BindingParamTest {
...
-        System.assertEquals( null, response.dateValue );
-        System.assertEquals( null, response.dateTimeValue );
-
-        System.assertEquals( 0, response.longValues.size() );
-        System.assertEquals( 0, response.decimalValues.size() );
-        System.assertEquals( 0, response.stringValues.size() );
-        System.assertEquals( 0, response.booleanValues.size() );
-        System.assertEquals( 0, response.dateValues.size() );
-        System.assertEquals( 0, response.dateTimeValues.size() );
+        System.Assert.isNull(response.longValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:47]: @chazwatkins -- @daveespo and I were discussing these changes.  One thing that Dave noticed was that the `isNull` method's error message actually doesn't express what he condition value is if it is not null.  The older `assertEquals` method error message does express what the value is.

We would like a change, please, in these instances where the no parameter `isNull` method is used to instead be the `System.Assert.areEqual(null, Object)` method.  

Please let me know if you have questions.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 2]
```diff
@@ -126,12 +126,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( 42, response.longValue );
-        System.assertEquals( 42.0, response.decimalValue );
-        System.assertEquals( '42', response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assertEquals( DateTime.newInstance(42).date(), response.dateValue );
-        System.assertEquals( DateTime.newInstance(42), response.dateTimeValue );
+        System.Assert.areEqual( 42, response.longValue );
+        System.Assert.areEqual( 42.0, response.decimalValue );
+        System.Assert.areEqual( '42', response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:132]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 3]
```diff
@@ -140,12 +140,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( 1, response.longValue );
-        System.assertEquals( 1.0, response.decimalValue );
-        System.assertEquals( '1', response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assertEquals( DateTime.newInstance(1).date(), response.dateValue );
-        System.assertEquals( DateTime.newInstance(1), response.dateTimeValue );
+        System.Assert.areEqual( 1, response.longValue );
+        System.Assert.areEqual( 1.0, response.decimalValue );
+        System.Assert.areEqual( '1', response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:146]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 4]
```diff
@@ -154,12 +154,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( 0, response.longValue );
-        System.assertEquals( 0, response.decimalValue );
-        System.assertEquals( '0', response.stringValue );
-        System.assertEquals( false, response.booleanValue );
-        System.assertEquals( DateTime.newInstance(0).date(), response.dateValue );
-        System.assertEquals( DateTime.newInstance(0), response.dateTimeValue );
+        System.Assert.areEqual( 0, response.longValue );
+        System.Assert.areEqual( 0, response.decimalValue );
+        System.Assert.areEqual( '0', response.stringValue );
+        System.Assert.areEqual( false, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:160]: Please change to `System.Assert.isFalse` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 5]
```diff
@@ -168,12 +168,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( expectedDateMillisGMT, response.longValue );
-        System.assertEquals( expectedDateMillisGMT, response.decimalValue );
-        System.assertEquals( String.valueOf( expectedDateMillisGMT ), response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assert( DateTime.newInstance( expectedDateMillisGMT ).date().isSameDay( response.dateValue ) );
-        System.assertEquals( DateTime.newInstance( expectedDateMillisGMT ), response.dateTimeValue );
+        System.Assert.areEqual( expectedDateMillisGMT, response.longValue );
+        System.Assert.areEqual( expectedDateMillisGMT, response.decimalValue );
+        System.Assert.areEqual( String.valueOf( expectedDateMillisGMT ), response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:174]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 6]
```diff
@@ -182,12 +182,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( expectedDateTimeMillisGMT, response.longValue );
-        System.assertEquals( expectedDateTimeMillisGMT, response.decimalValue );
-        System.assertEquals( String.valueOf( expectedDateTimeMillisGMT ), response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assert( DateTime.newInstance( expectedDateTimeMillisGMT ).date().isSameDay( response.dateValue ) );
-        System.assertEquals( DateTime.newInstance( expectedDateTimeMillisGMT ), response.dateTimeValue );
+        System.Assert.areEqual( expectedDateTimeMillisGMT, response.longValue );
+        System.Assert.areEqual( expectedDateTimeMillisGMT, response.decimalValue );
+        System.Assert.areEqual( String.valueOf( expectedDateTimeMillisGMT ), response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:188]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 7]
```diff
@@ -238,12 +238,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( null, response.longValue );
-        System.assertEquals( null, response.decimalValue );
-        System.assertEquals( inputs.get('text'), response.stringValue );
-        System.assertEquals( false, response.booleanValue );
-        System.assertEquals( null, response.dateValue );
-        System.assertEquals( null, response.dateTimeValue );
+        System.Assert.isNull(response.longValue );
+        System.Assert.isNull(response.decimalValue );
+        System.Assert.areEqual( inputs.get('text'), response.stringValue );
+        System.Assert.areEqual( false, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:244]: Please change to `System.Assert.isFalse` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 8]
```diff
@@ -252,12 +252,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( 42, response.longValue );
-        System.assertEquals( 42.0, response.decimalValue );
-        System.assertEquals( inputs.get('number'), response.stringValue );
-        System.assertEquals( false, response.booleanValue );
-        System.assertEquals( null, response.dateValue );
-        System.assertEquals( null, response.dateTimeValue );
+        System.Assert.areEqual( 42, response.longValue );
+        System.Assert.areEqual( 42.0, response.decimalValue );
+        System.Assert.areEqual( inputs.get('number'), response.stringValue );
+        System.Assert.areEqual( false, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:258]: Please change to `System.Assert.isFalse` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 9]
```diff
@@ -266,12 +266,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( null, response.longValue );
-        System.assertEquals( null, response.decimalValue );
-        System.assertEquals( inputs.get('truthy'), response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assertEquals( null, response.dateValue );
-        System.assertEquals( null, response.dateTimeValue );
+        System.Assert.isNull(response.longValue );
+        System.Assert.isNull(response.decimalValue );
+        System.Assert.areEqual( inputs.get('truthy'), response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:272]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 10]
```diff
@@ -280,12 +280,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( null, response.longValue );
-        System.assertEquals( null, response.decimalValue );
-        System.assertEquals( inputs.get('falsey'), response.stringValue );
-        System.assertEquals( false, response.booleanValue );
-        System.assertEquals( null, response.dateValue );
-        System.assertEquals( null, response.dateTimeValue );
+        System.Assert.isNull(response.longValue );
+        System.Assert.isNull(response.decimalValue );
+        System.Assert.areEqual( inputs.get('falsey'), response.stringValue );
+        System.Assert.areEqual( false, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:286]: Please change to `System.Assert.isFalse` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 11]
```diff
@@ -294,12 +294,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( null, response.longValue );
-        System.assertEquals( null, response.decimalValue );
-        System.assertEquals( inputs.get('date'), response.stringValue );
-        System.assertEquals( false, response.booleanValue );
-        System.assertEquals( testDateTime.date(), response.dateValue );
-        System.assertEquals( testDateTime.date(), response.dateTimeValue.dateGmt() ); // hours will be zero'ed out
+        System.Assert.isNull(response.longValue );
+        System.Assert.isNull(response.decimalValue );
+        System.Assert.areEqual( inputs.get('date'), response.stringValue );
+        System.Assert.areEqual( false, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:300]: Please change to `System.Assert.isFalse` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 12]
```diff
@@ -308,12 +308,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( null, response.longValue );
-        System.assertEquals( null, response.decimalValue );
-        System.assertEquals( inputs.get('dateTime'), response.stringValue );
-        System.assertEquals( false, response.booleanValue );
-        System.assertEquals( testDateTime.date(), response.dateValue );
-        System.assertEquals( testDateTime, response.dateTimeValue );
+        System.Assert.isNull(response.longValue );
+        System.Assert.isNull(response.decimalValue );
+        System.Assert.areEqual( inputs.get('dateTime'), response.stringValue );
+        System.Assert.areEqual( false, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:314]: Please change to `System.Assert.isFalse` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 13]
```diff
@@ -352,12 +352,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( 1, response.longValue );
-        System.assertEquals( 1.0, response.decimalValue );
-        System.assertEquals( 'true', response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assertEquals( null, response.dateValue );
-        System.assertEquals( null, response.dateTimeValue );
+        System.Assert.areEqual( 1, response.longValue );
+        System.Assert.areEqual( 1.0, response.decimalValue );
+        System.Assert.areEqual( 'true', response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:358]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 14]
```diff
@@ -366,12 +366,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( 0, response.longValue );
-        System.assertEquals( 0, response.decimalValue );
-        System.assertEquals( 'false', response.stringValue );
-        System.assertEquals( false, response.booleanValue );
-        System.assertEquals( null, response.dateValue );
-        System.assertEquals( null, response.dateTimeValue );
+        System.Assert.areEqual( 0, response.longValue );
+        System.Assert.areEqual( 0, response.decimalValue );
+        System.Assert.areEqual( 'false', response.stringValue );
+        System.Assert.areEqual( false, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:372]: Please change to `System.Assert.isFalse` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 15]
```diff
@@ -412,12 +412,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( testDateTime.getTime(), response.longValue );
-        System.assertEquals( testDateTime.getTime(), response.decimalValue );
-        System.assertEquals( String.valueOf( testDate ), response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assertEquals( testDate, response.dateValue );
-        System.assertEquals( testDateTime, response.dateTimeValue );
+        System.Assert.areEqual( testDateTime.getTime(), response.longValue );
+        System.Assert.areEqual( testDateTime.getTime(), response.decimalValue );
+        System.Assert.areEqual( String.valueOf( testDate ), response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:418]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_BindingParamTest.cls [hunk 16]
```diff
@@ -458,12 +458,12 @@ private class di_BindingParamTest {
...
-        System.assertEquals( testDateTime.getTime(), response.longValue );
-        System.assertEquals( testDateTime.getTime(), response.decimalValue );
-        System.assertEquals( String.valueOf( testDateTime ), response.stringValue );
-        System.assertEquals( true, response.booleanValue );
-        System.assertEquals( testDateTime.date(), response.dateValue );
-        System.assertEquals( testDateTime, response.dateTimeValue );
+        System.Assert.areEqual( testDateTime.getTime(), response.longValue );
+        System.Assert.areEqual( testDateTime.getTime(), response.decimalValue );
+        System.Assert.areEqual( String.valueOf( testDateTime ), response.stringValue );
+        System.Assert.areEqual( true, response.booleanValue );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_BindingParamTest.cls:464]: Please change to `System.Assert.isTrue` instead.

File: force-di/test/classes/di_InjectorComponentControllerTest.cls [hunk 17]
```diff
@@ -69,9 +69,8 @@ private class di_InjectorComponentControllerTest {
 
         Object result= newObj.getInject();
         //Then
-        System.assertNotEquals(null, result );
-        System.assertEquals(true,result instanceOf Component.apex.pageblock);
-
+        System.Assert.isNotNull(result );
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_InjectorComponentControllerTest.cls:72]: Please switch to `System.Assert.areNotEqual`

File: force-di/test/classes/di_PlatformCacheTest.cls [hunk 18]
```diff
@@ -41,7 +41,7 @@ private with sharing class di_PlatformCacheTest {
         // Given/ When
         di_Configurations__c config = di_PlatformCache.getConfig();
         //Then
-        System.assertEquals(true, config!=null);
+        System.Assert.isNotNull(config);
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_PlatformCacheTest.cls:44]: Please switch to `System.Assert.areNotEqual`

File: force-di/test/classes/di_PlatformCacheTest.cls [hunk 19]
```diff
@@ -197,8 +197,8 @@ private with sharing class di_PlatformCacheTest {
         // When
         cache.addBindingToKeyIndex(binding);
         // Then
-        system.assertEquals(1, cache.getCacheKeyIndexMap().size());
-        system.assertNotEquals(null,cache.getCacheKeyIndexMap().get(workingDeveloperName));
+        System.Assert.areEqual(1, cache.getCacheKeyIndexMap().size());
+        System.Assert.isNotNull(cache.getCacheKeyIndexMap().get(workingDeveloperName));
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_PlatformCacheTest.cls:201]: please switch to `System.Assert.areNotEqual`

File: force-di/test/classes/di_PlatformCacheTest.cls [hunk 20]
```diff
@@ -258,9 +258,9 @@ private with sharing class di_PlatformCacheTest {
         System.runAs(ruser) {
             // Then
             di_Configurations__c config = di_PlatformCache.getConfig();
-            system.assertEquals(true, config!=null);
-            system.assertEquals(false,  config.UsePlatformCacheToStoreBindings__c);
-            system.assertEquals(null, config.OrgCachePartitionName__c);
+            System.Assert.isNotNull(config);
+            System.Assert.areEqual(false,  config.UsePlatformCacheToStoreBindings__c);
+            System.Assert.isNull(config.OrgCachePartitionName__c);
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_PlatformCacheTest.cls:263]: Please switch to `System.Assert.areEqual`

File: force-di/test/classes/di_PlatformCacheTest.cls [hunk 21]
```diff
@@ -258,9 +258,9 @@ private with sharing class di_PlatformCacheTest {
         System.runAs(ruser) {
             // Then
             di_Configurations__c config = di_PlatformCache.getConfig();
-            system.assertEquals(true, config!=null);
-            system.assertEquals(false,  config.UsePlatformCacheToStoreBindings__c);
-            system.assertEquals(null, config.OrgCachePartitionName__c);
+            System.Assert.isNotNull(config);
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_PlatformCacheTest.cls:261]: Please switch to `System.Assert.areNotEqual`

File: force-di/test/classes/di_PlatformCacheTest.cls [hunk 22]
```diff
@@ -278,9 +278,9 @@ private with sharing class di_PlatformCacheTest {
         System.runAs(ruser) {
             // Then
             di_Configurations__c config = di_PlatformCache.getConfig();
-            system.assertEquals(true, config!=null);
-            system.assertEquals(true,  config.UsePlatformCacheToStoreBindings__c);
-            system.assertEquals(PARTITION, config.OrgCachePartitionName__c);
+            System.Assert.isNotNull(config);
```
Comment (ImJohnMDaniel) [inline force-di/test/classes/di_PlatformCacheTest.cls:281]: Please switch to `System.Assert.areNotEqual`

**Review by ImJohnMDaniel — CHANGES_REQUESTED** [review]

File: force-di/test/classes/di_BindingParamTest.cls (hunk 1 above)
Comment (chazwatkins) [reply force-di/test/classes/di_BindingParamTest.cls:47]: Sure, I'll make the updates on all the repos

**Comment by chazwatkins** [comment]
Thanks for reviewing the changes.  I made the requested updates.

- Assert.areEqual(true, ...) -> Assert.isTrue(...)
- Assert.areEqual(false, ...) -> Assert.isFalse(...)
- Assert.isNull(...) -> Assert.areEqual(null, ...)
- Assert.isNotNull(...) -> Assert.areNotEqual(null, ...)

**Review by ImJohnMDaniel — APPROVED** [review]

**Review by stohn777 — APPROVED** [review]

**Pull request closed without merging**